### PR TITLE
expose DPI and GUI scaling factors in prefs

### PR DESCRIPTION
**Pull request by aurelienpierre**
try to redraw the whole GUI when pref window is deleted, but only some widgets are redrawn.

Capture d’écran du 2020-06-14 18-17-06

### CODE REVIEW

File: src/gui/preferences.c [hunk 1]
```diff
@@ -379,6 +393,33 @@ static void init_tab_general(GtkWidget *stack)
   gtk_spin_button_set_value(GTK_SPIN_BUTTON(fontsize), dt_conf_get_float("font_size"));
   g_signal_connect(G_OBJECT(fontsize), "value_changed", G_CALLBACK(font_size_changed_callback), 0);
 
+  GtkWidget *screen_ppd_overwrite = gtk_spin_button_new_with_range(-1.0f, 8.0f, 0.2f);
+  label = gtk_label_new(_("images DPI scaling factor"));
```
Comment (TurboGit) [inline src/gui/preferences.c:397]: images -> thumbs

Which will be less ambiguous I think.

**Comment by TurboGit** [comment]
@aurelienpierre : are you targeting this for 3.2 ?

**Comment by aurelienpierre** [comment]
> @aurelienpierre : are you targeting this for 3.2 ?

Yes, it's long due and has been requested many many times.

File: src/gui/preferences.c (hunk 1 above)
Comment (aurelienpierre) [reply src/gui/preferences.c]: done, I have also reworked the other label to make the difference more obvious

**Comment by TurboGit** [comment]
Hum... I set GUI DPI to 96 then GUI scaling factor to 1.2, leaving the entry the 1.2 was changed in 8.0. Exited the preferences and dt was not responding, my computer started to set ventilator very fast I had to kill dt but it was quite difficult as it was eating most of my CPU ! 

I would say not ready for integration :)

**Comment by parafin** [comment]
I wonder if non-integer PPD (1.2) can cause troubles, or does it make sense at all.

**Comment by coffeeandspaceships** [comment]
Below is from `darktable.c`. ppd is used in a few other places where float seems ok; but  I noticed this one carries out a multiply without a cast to int. That will round 1.2 down to 1.0 again if the compiler (and flags applied to compiler...) support implicit conversation, I think? 
```    // blow up the PNG. Ugly, but at least it has the correct size afterwards :-/
    const int width = DT_PIXEL_APPLY_DPI(png_width) * darktable.gui->ppd,
              height = DT_PIXEL_APPLY_DPI(png_height) * darktable.gui->ppd;
    const int stride = cairo_format_stride_for_width(CAIRO_FORMAT_ARGB32, width);
```

**Comment by parafin** [comment]
Your analysis is incorrect:

- first of all there is no "if the compiler supports" - all compilers follow the language standard and behave the same, at least in this regard
- DT_PIXEL_APPLY_DPI multiplies int by double, result value is double (int is converted to double)
- conversion back to int happens after calculation is done (multiplication of double by double)

**Comment by aurelienpierre** [comment]
> Hum... I set GUI DPI to 96 then GUI scaling factor to 1.2, leaving the entry the 1.2 was changed in 8.0. Exited the preferences and dt was not responding, my computer started to set ventilator very fast I had to kill dt but it was quite difficult as it was eating most of my CPU !

I tried to be clever and have new sizes loaded and GUI redrawn when exiting the pref window. First of all, not everything is redrawn(although redrawing main window should redraw all children too…), but it could be the source of the problems. 

Maybe just ask user to restart ?

**Comment by TurboGit** [comment]
Ok, does not crash my DE now :) Which is good... Merging!

**Comment by TurboGit** [comment]
Thanks !

**Pull request merged**